### PR TITLE
Improve release process

### PR DESCRIPTION
**Pull request by uncleDecart**
This PR is RFC. It introduces doc, describing release process and changes to gha to publish release to ghcr.io as well as to dockerhub. The process described in the doc is similar to what we are doing in Eden and EVE, that way we can be consistent across repos

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -20,17 +25,25 @@ jobs:
...
-        uses: docker/login-action@v1 
+        uses: docker/login-action@v2
         with:
           username: ${{ secrets.RELEASE_DOCKERHUB_ACCOUNT }}
           password: ${{ secrets.RELEASE_DOCKERHUB_TOKEN }}
-      - name: Build and push
+      - name: Build and push to Dockerhub
         id: docker_build
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v5
```
Comment (deitch) [inline .github/workflows/release.yaml:34]: Save the trouble of building it multiple times. Just have multiple tags for all of them, including the ghcr.io tags. Sort of:

```yaml
      - name: Build and push
        id: docker_build
        uses: docker/build-push-action@v5
        with:
          platforms: linux/amd64,linux/arm64
          push: true
          tags: |
            ${{env.IMAGE_NAME}}:${{env.GITHUB_REF#refs/tags/}}
            ${{env.IMAGE_NAME}}:latest
            ghcr.io/${{ github.repository_owner }}/adam:${{ github.event.release.tag_name }}
            ghcr.io/${{ github.repository_owner }}/adam:latest
```

**Review by deitch — CHANGES_REQUESTED** [review]
Minor change to simplify the flow.

Thank you especially for the docs.

**Review by deitch — APPROVED** [review]

**Comment by deitch** [comment]
LGTM. Ready to merge?

**Comment by uncleDecart** [comment]
Yes, let's merge and then draft a release for v0.1.0 as a starting point

**Review by yash-zededa — APPROVED** [review]

**Comment by shjala** [comment]
LGTM.

**Pull request merged**